### PR TITLE
feat: (Gen AI): Add a sample for updating context cache with an expiration date

### PR DESCRIPTION
**Pull request by Valeriy-Burlaka**
## Description

Adds a new sample for the "Update the context cache using its expire_time parameter"  [section](https://cloud.google.com/vertex-ai/generative-ai/docs/context-cache/context-cache-update#update_the_context_cache_using_its_expire_time_parameter) of the `context-cache/context-cache-update` Vertex AI [page](https://cloud.google.com/vertex-ai/generative-ai/docs/context-cache/context-cache-update).

## Checklist
- [X] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [X] **Tests** pass:   `nox -s py-3.8` and `nox -s py-3.11`
- [X] **Lint** pass:   `nox -s lint` 
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by Valeriy-Burlaka** [comment]
The `py-3.11` build failed because of a ".RESOURCE_EXHAUSTED" / "out of capacity" Error in a different test:

```
=================================== FAILURES ===================================
_________________________ test_config_response_schema7 _________________________
Traceback (most recent call last):
  File "/workspace/generative_ai/.nox/py-3-11/lib/python3.11/site-packages/google/api_core/grpc_helpers.py", line 76, in error_remapped_callable
    return callable_(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspace/generative_ai/.nox/py-3-11/lib/python3.11/site-packages/grpc/_channel.py", line 1181, in __call__
    return _end_unary_response_blocking(state, call, False, None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspace/generative_ai/.nox/py-3-11/lib/python3.11/site-packages/grpc/_channel.py", line 1006, in _end_unary_response_blocking
    raise _InactiveRpcError(state)  # pytype: disable=not-instantiable
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.RESOURCE_EXHAUSTED
	details = "Unable to submit request because the service is temporarily out of capacity. Try again later."
	debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.183.95:443 {grpc_message:"Unable to submit request because the service is temporarily out of capacity. Try again later.", grpc_status:8, created_time:"2024-09-25T18:11:53.078829516+00:00"}"
	```

File: generative_ai/context_caching/update_context_cache.py [hunk 1]
```diff
@@ -31,8 +33,13 @@ def update_context_cache(cache_id: str) -> str:
 
     cached_content = caching.CachedContent(cached_content_name=cache_id)
 
-    # Update the expiration time by 1 hour
-    cached_content.update(ttl=datetime.timedelta(hours=1))
+    # Update the context cache to expire in 3 hours
```
Comment (msampathkumar) [inline generative_ai/context_caching/update_context_cache.py:36]:
```suggestion
    # Option1: Update the context cache using ttl(time to live)
```

File: generative_ai/context_caching/update_context_cache.py [hunk 2]
```diff
@@ -31,8 +33,13 @@ def update_context_cache(cache_id: str) -> str:
...
+    cached_content.update(ttl=timedelta(hours=3))
+    cached_content.refresh()
+
+    # We can also set the cache expiration to a specific time in the future
```
Comment (msampathkumar) [inline generative_ai/context_caching/update_context_cache.py:40]:
```suggestion
    # Option1: Update the context cache using specific time
```

**Review by msampathkumar — COMMENTED** [review]
Overall PR looks good. Need minor changes.

**Review by msampathkumar — APPROVED** [review]
LGTM. Nice work @Valeriy-Burlaka

**Pull request merged**